### PR TITLE
[Oy2-18726] Measure Kebab Button

### PR DESCRIPTION
**Pull request by jphechter**
## Purpose

Add functionality to individual measure's kebab menu on the CoreSet view.

#### Linked Issues to Close

[Oy2-18726](https://qmacbis.atlassian.net/browse/OY2-18726?atlOrigin=eyJpIjoiNzNmYzZjMDdkYTZlNGIxYWE5MzA0YzBkZjliZjIzZjAiLCJwIjoiaiJ9)

## Approach

- Change EDIT to VIEW
- VIEW option drives to the corresponding measure form
- Create an EDIT option for user created measures
- EDIT option opens a modal that allows the user to update the measure meta-data (description and detailedDescription)
- UpdateMeasure endpoint optionally accounts for changes to description and detailedDescription

Screen Shot 2022-06-16 at 2 29 05 PM

#### Pull Request Creator Checklist

- [X] This PR has an associated issue or issues.
- [X] The associated issue(s) are linked above.
- [X] This PR meets all acceptance criteria for those issues.
- [X] This PR and linked issue(s) are adequately documented
- [X] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [X] Someone has been assigned this PR.
- [X] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/ui-src/src/components/UpdateInfoModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                  name="update-ssm.description"
+                  rules={{ required: true }}
+                ></QMR.TextInput>
+                <QMR.TextArea
+                  label="Please provide a description of the measure"
+                  name="update-ssm.detailedDescription"
+                  rules={{ required: true }}
+                ></QMR.TextArea>
+              </CUI.VStack>
+            </FormProvider>
```
Comment (jphechter) [inline services/ui-src/src/components/UpdateInfoModal/index.tsx:99]: This is almost a direct copy from `AddSSM/index.tsx`. Could be extracted into its own component and used in both places.

File: services/ui-src/src/views/CoreSet/index.tsx [hunk 2]
```diff
@@ -342,6 +344,7 @@ export const CoreSet = () => {
         </CUI.Box>
       </CUI.Flex>
       <CUI.Box mt="4">
+        <p>TEst this</p>
```
Comment (charmcitygavin) [inline services/ui-src/src/views/CoreSet/index.tsx]: Is this leftover test text or are we using it for something?

File: services/ui-src/src/components/UpdateInfoModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        description,
+        detailedDescription,
+      },
+    });
+  }, [description, detailedDescription, methods]);
+
+  /*
+   * Return boolean value determining if modal data can be submitted
+   */
+  const isDisabled = () => {
```
Comment (benjaminpaige) [inline services/ui-src/src/components/UpdateInfoModal/index.tsx:50]: i dont think this needs to be a function, something like

```
isDisabled =
!(watchedDescription && watchedDetailedDescription) ||
(watchedDescription === description &&
 watchedDetailedDescription === detailedDescription)
```

File: services/ui-src/src/components/UpdateInfoModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+              colorScheme="blue"
+              mr={3}
+              isDisabled={isDisabled()}
+              onClick={() => {
+                handleModalResponse(modalResponseData());
+              }}
+            >
+              Yes
+            </CUI.Button>
+            <CUI.Button variant="ghost" onClick={() => closeModal()}>
```
Comment (benjaminpaige) [inline services/ui-src/src/components/UpdateInfoModal/index.tsx:112]: could be just `onClick={closeModal}` but either way

File: services/ui-src/src/views/CoreSet/AddSSMCard/index.tsx [hunk 5]
```diff
@@ -36,7 +36,7 @@ export const AddSSMCard = ({
           <CUI.Text fontWeight="bold">{title}</CUI.Text>
           <Link
             to={to}
-            state={{ userCreatedMeasureIds: userCreatedMeasureIds }}
+            state={{ userCreatedMeasureIds }}
```
Comment (abelzer212) [inline services/ui-src/src/views/CoreSet/AddSSMCard/index.tsx:39]: i love it

File: services/ui-src/src/components/UpdateInfoModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+              </CUI.VStack>
+            </FormProvider>
+          </CUI.ModalBody>
+          <CUI.ModalFooter>
+            <CUI.Button
+              colorScheme="blue"
+              mr={3}
+              isDisabled={isDisabled()}
+              onClick={() => {
+                handleModalResponse(modalResponseData());
```
Comment (benjaminpaige) [inline services/ui-src/src/components/UpdateInfoModal/index.tsx:107]: maybe just `onClick={handleUpdateInfo}` and then call handleModalResponse in the handleUpdateInfo function with a data object?

File: services/ui-src/src/views/CoreSet/index.tsx [hunk 7]
```diff
@@ -167,13 +183,21 @@ const useMeasureTableDataBuilder = () => {
       const measureTableData = (filteredItems as MeasureData[]).map((item) => {
         const actions = [
           {
-            itemText: "Edit",
-            handleSelect: () => console.log("Edit " + item.measure),
+            itemText: "View",
+            handleSelect: () => {
+              window.location.href = `${window.location.href}/${item.measure}`;
```
Comment (benjaminpaige) [inline services/ui-src/src/views/CoreSet/index.tsx:188]: just use navigate
```
import { useNavigate } from "react-router-dom"
const navigate = useNavigate();
navigate(`${item.state}/${item.year}/${item.measure}`)
```

or something like that

File: services/ui-src/src/views/CoreSet/index.tsx (hunk 2 above)
Comment (jphechter) [reply services/ui-src/src/views/CoreSet/index.tsx]: 100% leftovers

File: services/ui-src/src/components/UpdateInfoModal/index.tsx (hunk 3 above)
Comment (jphechter) [reply services/ui-src/src/components/UpdateInfoModal/index.tsx:50]: I had originally had the check like this and extrapolated it for readability. With shorter variable names this is more digestible.

File: services/ui-src/src/components/UpdateInfoModal/index.tsx (hunk 4 above)
Comment (jphechter) [reply services/ui-src/src/components/UpdateInfoModal/index.tsx:112]: Yes, can do.

File: services/ui-src/src/components/UpdateInfoModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  }, [description, detailedDescription, methods]);
+
+  /*
+   * Return boolean value determining if modal data can be submitted
+   */
+  const isDisabled = () => {
+    return (
+      // Fields must not be empty
+      !(
+        methods.watch()?.["update-ssm"]?.description &&
```
[the remark this answers is not in the document]
Comment (jphechter) [reply services/ui-src/src/components/UpdateInfoModal/index.tsx:54]: Yup that works.

File: services/ui-src/src/components/UpdateInfoModal/index.tsx (hunk 6 above)
Comment (jphechter) [reply services/ui-src/src/components/UpdateInfoModal/index.tsx:107]: Yup that works.

File: services/ui-src/src/components/UpdateInfoModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+  "update-ssm": MeasureDetail;
+}
+
+export const UpdateInfoModal = ({
+  handleModalResponse,
+  closeModal,
+  modalProps: {
+    isOpen,
+    measure,
+    measure: { description, detailedDescription },
```
Comment (benjaminpaige) [inline services/ui-src/src/components/UpdateInfoModal/index.tsx:28]: does this just get over written?

**Review by benjaminpaige — APPROVED** [review]
love it!

**Review by abelzer212 — APPROVED** [review]

**Pull request merged**